sdk/provision/file_provisioner.go

```diff
@@ -159,7 +159,7 @@ func (p FileProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, o
 			argsResolved[i] = result.String()
 		}
 
-		out.AddArgs(argsResolved...)
+		out.PrependArgs(argsResolved...)
```

Comment:
Do we want to do this unconditionally or just for `mysql`? We should verify this doesn't break any of the existing plugins first.

Reply:
Fair enough, though it's kind of unique to the `mysql` CLI that flag-type arguments be order-sensitive.

It's been a long while since I made this PR but as I recall, scoping the change to a single plugin was more involved.  I will do what I can but may not have any time in the near future to fix up my PR limit the argument ordering change to a single plugin.

Reply:
It may be fine as is. We just need to verify it doesn’t break anything.